bittide/src/Bittide/Axi4.hs

```diff
@@ -0,0 +1,105 @@
+-- SPDX-FileCopyrightText: 2023 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+{-# OPTIONS_GHC -fconstraint-solver-iterations=10 #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Bittide.Axi4 where
+
+import Clash.Prelude
+
+import Data.Maybe
+import Protocols.Axi4.Stream
+
+type EndOfPacket = Bool
+type BufferFull = Bool
+
+data WbAxisRxBufferState fifoDepth nBytes = WbAxisRxBufferState
+  { readingFifo :: Bool
+  , packetLength :: Index (fifoDepth * nBytes + 1)
+  , writeCounter :: Index fifoDepth
+  , packetComplete :: Bool
+  , bufferFull :: Bool
+  } deriving (Generic, NFDataX)
+
+{-# NOINLINE axisFromByteStream #-}
+axisFromByteStream ::
+  forall dom confA confB userType .
+  ( HiddenClockResetEnable dom, DataWidth confA ~ 1, KnownAxi4StreamConfig confB
+  , 1 <= DataWidth confB
+  , DestWidth confA ~ DestWidth confB
+  , IdWidth confA ~ IdWidth confB
+  , Eq userType, NFDataX userType) =>
```

Comment:
```suggestion
  ( HiddenClockResetEnable dom
  , KnownAxi4StreamConfig confB
  , DataWidth confA ~ 1
  , 1 <= DataWidth confB
  , DestWidth confA ~ DestWidth confB
  , IdWidth confA ~ IdWidth confB
  , Eq userType
  , NFDataX userType) =>
```

The mix/match between multiple and single constraints on a single line makes it hard to read.